leftwm-core/src/models/layout_manager.rs

```diff
@@ -19,40 +19,53 @@ impl Default for LayoutMode {
 pub struct LayoutManager {
     pub mode: LayoutMode,
     pub layouts: Vec<Layout>,
+    pub layouts_per_workspaces: HashMap<i32, Vec<Layout>>,
 }
 
 impl LayoutManager {
     pub fn new(config: &impl Config) -> Self {
+        let layouts_per_workspaces = config
+            .workspaces()
+            .unwrap_or_default()
+            .iter()
+            .filter_map(|w| w.id.map(|id| (id, w.layouts.clone())))
+            .collect();
+
         Self {
             mode: config.layout_mode(),
             layouts: config.layouts(),
+            layouts_per_workspaces,
         }
     }
 
     pub fn new_layout(&self) -> Layout {
         *self.layouts.first().unwrap_or(&Layout::default())
     }
 
-    pub fn next_layout(&self, layout: Layout) -> Layout {
-        let mut index = match self.layouts.iter().position(|&x| x == layout) {
-            Some(x) => x as isize,
-            None => return Layout::default(),
-        } + 1;
-        if index >= self.layouts.len() as isize {
-            index = 0;
-        }
-        self.layouts[index as usize]
+    pub fn next_layout(&self, layout: Layout, workspace: &Workspace) -> Layout {
```

Comment:
Should we just pass the id e.g.:
```rs
pub fn next_layout(...., id: Option<i32>)...
```

Reply:
we could also pass the workspace itself, since everywhere this is called, the layout comes from the workspace anyway